### ISSUE TITLE
Adapt to "lens" library removing defaultRules in ver. 4.4 (was: building Ganeti 2.12 on CentOS7)

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#1037]
Originally reported of Google Code with ID 981.

```
What software version are you running? Please provide the output of "gnt-
cluster --version", "gnt-cluster version", and "hspace --version".

<b>What distribution are you using?</b>
CentOS 7
ghc 7.6 + cabal
stable-2.12 from source

<b>What steps will reproduce the problem?</b>
1. ./configure --with-user-prefix='ganeti-' --with-group-prefix='ganeti-' 
2. make

<b>What is the expected output? What do you see instead?</b>
/bin/ghc -O -Wall -isrc -fwarn-monomorphism-restriction -fwarn-tabs -fwarn-incomplete-uni-patterns    \
         \
        -DPARALLEL3    --make src/ganeti-kvmd.hs
Linking src/ganeti-kvmd ...
[GHC]: src/Ganeti/Utils/MVarLock.o <- src/Ganeti/Utils/MVarLock.hs
[GHC]: src/Ganeti/Query/Types.o <- src/Ganeti/Query/Types.hs src/Ganeti/Objects.hi src/Ganeti/Objects.o src/Ganeti/Query/Language.hi src/Ganeti/Query/Language.o
[GHC]: src/Ganeti/Query/Filter.o <- src/Ganeti/Query/Filter.hs src/Ganeti/JSON.hi src/Ganeti/JSON.o src/Ganeti/Query/Types.hi src/Ganeti/Query/Types.o src/Ganeti/Query/Language.hi src/Ganeti/Query/Language.o src/Ganeti/Objects.hi src/Ganeti/Objects.o src/Ganeti/Errors.hi src/Ganeti/Errors.o src/Ganeti/BasicTypes.hi src/Ganeti/BasicTypes.o
[GHC]: src/Ganeti/Lens.o <- src/Ganeti/Lens.hs

src/Ganeti/Lens.hs:68:23: Not in scope: `defaultRules'
make: *** [src/Ganeti/Lens.o] Error 1
make  174.92s user 11.36s system 99% cpu 3:07.06 total


<b>Please provide any additional information below.</b>

```
Originally added on 2014-10-27 18:46:52 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2014-10-28 10:08:16 +0000 UTC.

Added Labels: Type-Defect Priority-Medium HaskellTask
Changed State: Accepted
Added to Milestone: Release2.12

**Comment by Ganeti-Issues-Migrator** [comment]
```
An issue for "lens" filed: https://github.com/ekmett/lens/issues/495
```
Originally added on 2014-10-28 11:39:48 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Fixed in 2.12.2, commits:

commit 14a85a6fa426e3088423923094cd6d574fe91d3f
Author: Petr Pudlak <pudlak@google.com>
Date:   Tue Feb 10 13:18:00 2015 +0100

    Construct lenses correctly for different 'lens' versions
    
    As the interface of the TH code for generating lenses changed twice in
    4.*, we need to differentiate between versions accordingly.
    
    Signed-off-by: Petr Pudlak <pudlak@google.com>
    Reviewed-by: Klaus Aehlig <aehlig@google.com>

commit cfd9c8a82550df4e29ebeee2158f1d7fb864d531
Author: Petr Pudlak <pudlak@google.com>
Date:   Wed Feb 11 13:45:37 2015 +0100

    Create a simple CPP macro for detecting lens versions
    
    .. in Haskell code.
    
    This patch should be removed starting from 2.14, as there cabal
    provides proper macros for all packages.
    
    The macro in this patch is intentionally kept compatible with the cabal
    macros.
    
    Signed-off-by: Petr Pudlak <pudlak@google.com>
    Reviewed-by: Klaus Aehlig <aehlig@google.com>
```
Originally added on 2015-05-21 08:17:45 +0000 UTC.

Changed State: Released
